### DISCUSSION TITLE
INTEGER_SCALE Default Off in tic80lua.wasm (v 1.1.2837)

### DISCUSSION THREAD

**Discussion by bitwisecreative** [nesbox/TIC-80#2386] [Q&A] [answered]
I built a custom version with INTEGER_SCALE_DEFAULT false (config.c) and it works for the binary but exporting to html with alone=1 does not result in integer scale default off. I can't figure out how/where the tic80lua.wasm alone html export file is built or how to set it to integer_scale default off.

**Comment by nesbox** [comment]
same story but with crt filter here #1882

**Reply by bitwisecreative** [reply]
Maybe I've over-complicated what I'm trying to do... I wanted integer scale default off for anyone who opened my game in the browser. If there's a way I could have done that without rebuilding the wasm files, well ****. I finally got it worked out last night with some help by rebuilding the wasm files and will post my solution below.

**Comment by bitwisecreative** [comment, marked as answer]
Thanks to @soxfox42 for helping me with this.

How I rebuilt the tic80lua.wasm and tic80lua.js files for version 1.1.2837

I already had my game exported without editors...

`export html game alone=1`

Follow init build instructions from wiki:

`sudo apt-get install g++ git cmake ruby-full libglvnd-dev libglu1-mesa-dev freeglut3-dev libasound2-dev -y`

`git clone --recursive https://github.com/nesbox/TIC-80`

`cd TIC-80`

I wanted to work with a specific version, so...

`git checkout be42d6f146cfa520b9b1050feba10cc8c14fb3bd`

Then I made my source code change in src/studio/config.c

`#define INTEGER_SCALE_DEFAULT false`

Install Emscripten SDK. (https://emscripten.org/docs/getting_started/downloads.html)

`cd build`

`emcmake cmake -DBUILD_SDLGPU=On -DCMAKE_BUILD_TYPE=MinSizeRel -DBUILD_STUB=On ..`

`cmake --build . --parallel`

Output is in build/bin as tic80lua.wasm and tic80lua.js

Replace the files in the export... 

`cp bin/tic80lua.wasm /exported/game/`
`cp bin/tic80lua.js /exported/game/tic80.js`
